flink-end-to-end-tests/flink-tpcds-test/src/main/java/org/apache/flink/table/tpcds/TpcdsTestProgram.java

```diff
@@ -107,23 +113,44 @@ public static void main(String[] args) throws Exception {
 
             // register sink table
             String sinkTableName = QUERY_PREFIX + queryId + "_sinkTable";
-            ((TableEnvironmentInternal) tableEnvironment)
-                    .registerTableSinkInternal(
-                            sinkTableName,
-                            new CsvTableSink(
-                                    sinkTablePath + FILE_SEPARATOR + queryId + RESULT_SUFFIX,
-                                    COL_DELIMITER,
-                                    1,
-                                    FileSystem.WriteMode.OVERWRITE,
-                                    resultTable.getSchema().getFieldNames(),
-                                    resultTable.getSchema().getFieldDataTypes()));
+            tableEnvironment.createTable(
+                    sinkTableName,
+                    getTableDescriptor(
+                            sinkTablePath + FILE_SEPARATOR + queryId + RESULT_SUFFIX,
+                            resultTable.getSchema().getFieldNames(),
+                            resultTable.getSchema().getFieldDataTypes()));
             TableResult tableResult = resultTable.executeInsert(sinkTableName);
             // wait job finish
             tableResult.getJobClient().get().getJobExecutionResult().get();
             System.out.println("[INFO]Run TPC-DS query " + queryId + " success.");
         }
     }
 
+    private static TableDescriptor getTableDescriptor(
+            String path, String[] fieldNames, DataType[] fieldDataTypes) {
+        final Schema.Builder schemaBuilder = Schema.newBuilder();
+        for (int i = 0; i < fieldNames.length; i++) {
+            schemaBuilder.column(fieldNames[i], fieldDataTypes[i]);
+        }
+        final Schema schema = schemaBuilder.build();
+
+        final FormatDescriptor formatDescriptor =
+                FormatDescriptor.forFormat("csv")
+                        .option(CsvFormatOptions.FIELD_DELIMITER, COL_DELIMITER)
+                        .build();
+
+        return TableDescriptor.forConnector(FileSystemTableFactory.IDENTIFIER)
+                .schema(schema)
+                .option(FileSystemConnectorOptions.PATH, path)
```

Comment:
I'm just curious that current write mode is insert into or insert overwrite?

Reply:
The old CSV sink writes to a specified file, so it needs to distinguish between `append` and `override` write modes. In contrast, the new filesystem connector writes to a directory, so it does not need to be aware of `append` or `override` write modes.